### DISCUSSION TITLE
add support for io_uring if improves gdu

### DISCUSSION THREAD

**Discussion by danie-dejager** [dundee/gdu#183] [Ideas]
@dundee Will it be possible to add support for io_uring? And if so won't gdu benefit from modern kernels supporting this?

https://unixism.net/loti/what_is_io_uring.html

This allows for io_uring in go https://github.com/Iceber/iouring-go

**Comment by dundee** [comment]
Yes, we can check if gdu can benefit from this.

**Reply by danie-dejager** [reply]
The newer `plocate` for instance is much faster than the older `mlocate`.

https://plocate.sesse.net/

**Comment by dundee** [comment]
I have tried to use io_uring in https://github.com/dundee/gdu/pull/195 but it's much slower than the current implementation for now. I will need to spend more time on it.
